### ISSUE TITLE
(RFE) DescriptionList: missing option where the horizonal list can become vertical on small screens

### ISSUE THREAD

**Issue by KKoukiou** [patternfly/patternfly-react#4777]
In cockpit we would like to use isHorizontal for description lists (1 column), but we would like that they are responsive and convert into vertical for smaller screen sizes.
It would be useful to provide an option for that.

So instead of boolean it would be nice to have: isHorizontal: { "lg" } or something similar.

**Comment by KKoukiou** [comment]
The desired behavior is what Form component has, https://www.patternfly.org/v4/documentation/react/components/form/horizontal
which wrap to vertical for very small screens.

**Comment by tlabaj** [comment]
cc @mcarrano @mcoker

**Comment by mcarrano** [comment]
@mattnolting thoughts?

**Comment by mcoker** [comment]
Seems like a good enhancement to me.

@mcarrano We can add modifier variations to specify a horizontal/vertical layout at different breakpoints. The horizontal form layout switches to a vertical layout below the medium breakpoint. Do you think the description list should do that automatically, too? Or would the user need to specify a variation for the layout to change between horizontal and vertical?

We can transfer this issue to core, or open a new one and update this issue once the core work is complete.

**Comment by mcarrano** [comment]
I'm going to move this over to Core in that case. @KKoukiou do you have a specific timeframe when this is needed?
